src/ripple/core/impl/Config.cpp

```diff
@@ -212,12 +216,12 @@ void Config::setup (std::string const& strConf, bool bQuiet)
 
     QUIET       = bQuiet;
 
-    strDbPath           = Helpers::getDatabaseDirName ();
-    strConfFile         = strConf.empty () ? Helpers::getConfigFileName () : strConf;
-
-    VALIDATORS_BASE     = Helpers::getValidatorsFileName ();
+    strDbPath = databaseDirName;
 
-    VALIDATORS_URI      = boost::str (boost::format ("/%s") % VALIDATORS_BASE);
+    if (!strConf.empty())
+        strConfFile = strConf;
+    else
+        strConfFile = configFileName;
```

Comment:
What's the return type of the ternary when one of the arguments is `char const* const` and the other is `std::string`?